internal/patch/legacy.go

```diff
@@ -0,0 +1,88 @@
+package patch
+
+import (
+	"encoding/json"
+	"fmt"
+
+	redsky "github.com/redskyops/redskyops-controller/api/v1beta1"
+	"github.com/redskyops/redskyops-controller/internal/template"
+	"github.com/redskyops/redskyops-controller/internal/trial"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const defaultAttemptsRemaining = 3
+
+// RenderTemplate determines the patch target and renders the patch template
+func RenderTemplate(te *template.Engine, t *redsky.Trial, p *redsky.PatchTemplate) (*corev1.ObjectReference, []byte, error) {
+	// Render the actual patch data
+	data, err := te.RenderPatch(p, t)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Determine the reference, possibly extracting it from the rendered data
+	ref := &corev1.ObjectReference{}
+	switch {
```

Comment:
I think this might be an agree to disagree bit. I'd really like to find a way to simplify this chunk of logic, but that's out of scope for this round.